src/json/from_json.cr

```diff
@@ -65,6 +65,57 @@ def Deque.from_json(string_or_io) : Nil
   end
 end
 
+module Iterator(T)
+  # Reads the content of a JSON array into an iterator in a lazy way.
+  # With this method it should be possible to process a huge JSON array, without
+  # the requirement that the whole array fits into memory.
+  #
+  # The following example produces a huge file, uses a lot of CPU but should not require much memory.
+  #
+  # ```
+  # File.open("/tmp/test.json", "w+") do |f|
+  #   (0..1_000_000_000).each.to_json(f)
+  # end
+  #
+  # File.open("/tmp/test.json", "r") do |f|
+  #   p Iterator(Int32).from_json(f).skip(1_000_000_000).to_a
+  # end
+  # ```
+  #
+  # WARNING: The `string_or_io` can't be used by anything else until the iterator is fully consumed.
+  def self.from_json(string_or_io)
+    Iterator(T).new(JSON::PullParser.new(string_or_io))
+  end
+
+  # Creates a new iterator which iterates over a JSON array. See also `Iterator#from_json`.
+  #
+  # WARNING: The `JSON::PullParser` can't be used by anything else until the iterator is fully consumed.
```

Comment:
I think this is generally true when you pass a pull parser. So I'm not sure we should include this warning. Like, a pull parser shouldn't have more than one owner.

Reply:
That said, we can remove these things later on.